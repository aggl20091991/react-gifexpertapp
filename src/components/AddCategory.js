import React, {useState} from 'react'
import PropTypes from 'prop-types'

const AddCategory = ({setCategories}) => {
    const [inputValue,setInputValue] = useState('');
    const handlerInputChange = (e) =>{
        setInputValue(e.target.value)
    }
    const handlerSubmit = (e) =>{
        //se usa para no dar el f5
        e.preventDefault();
        if(inputValue.trim().length>1){
        setCategories(cats =>[inputValue,...cats]);
        //se manda a vacio el input
        setInputValue('')
        }

    }
  return (
      <form onSubmit={handlerSubmit}>
     <input type="text" 
     value={inputValue} 
     onChange={handlerInputChange}
     
     />   
      </form>
  )
}
AddCategory.protoTypes = {
    setCategories : PropTypes.func.isRequired
}

export default AddCategory