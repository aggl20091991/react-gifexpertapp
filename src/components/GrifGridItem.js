import React from 'react'
import PropTypes from 'prop-types';

function GrifGridItem({id,title,url}) {
  return (
    <div className="card animate__animated animate__bounce">
        <img src={url} alt={title}/>
        <p>{title}</p>
    
    </div>
  )
}
GrifGridItem.prototypes = {
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
}

export default GrifGridItem