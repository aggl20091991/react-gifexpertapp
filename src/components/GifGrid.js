import React,{useState,useEffect} from 'react'
import GrifGridItem from './GrifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

function GifGrid({category}) {
  
 console.log(category);   
const {data:images,loading} = useFetchGifs( {category} );

return (
      <>
    <h3>{category}</h3>
    
<div className="card-grid">
        {
            //otra manera de enviar parametros a objetos
            images.map(img=>(
            <GrifGridItem key={img.id} {...img} />
        ))
        }
      
</div>
</>
  )
}

export default GifGrid