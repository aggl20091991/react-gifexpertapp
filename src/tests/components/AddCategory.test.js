import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { AddCategory } from '../../components/AddCategory';

describe('pruebas en <AddCategory />', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />)

    beforeEach(()=>{
        jest.clearAllMocks();
         wrapper = shallow(<AddCategory setCategories={setCategories} />)
    })

    test('creando snapshot', () => { 
        expect( wrapper ).toMatchSnapshot();
     })

     test('should be cambiar caja de texto', () => { 
         
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        console.log(wrapper);
        input.simulate('change',{target:{value:value}});
        
        expect(wrapper.find('p').text().trim()).toBe(value);


      })
      test('should no debe postear on submit', () => { 
        wrapper.find('form').simulate('submit',{
            preventDefault(){}
        });         
        
        expect( setCategories ).not.toHaveBeenCalled();

       })

       test('should limpiar caja de texto', () => { 
           const value='hola mundo';
        wrapper.find('input').simulate('change',{target:{value}});
        wrapper.find('form').simulate('submit',{ preventDefault(){}   });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(wrapper.find('input').prop('value')).toBe('');

       
    });   

})