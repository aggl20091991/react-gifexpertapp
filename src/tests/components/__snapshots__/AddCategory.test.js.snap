// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <AddCategory /> creando snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
